gohypergiant/hyperdrive
# MLSDK-412 Fastapp runtime hyperpack upload

## Summary

This PR adds a web UI at the root of the fastapp container for a user to upload a new hyperpack for the fastapp backend to use. 

## Testing

* From `/hyperdrive/fastapp`, run `make build-dev` and `make run-dev`
* Visit http://0.0.0.0:8001/, select a Hyperpack and upload it (ping me or John Hou to get a sample hyperpack)
* Verify it lands in /upload, and that you can run /sample-curl.sh and get this result: `{"prediction":0}`
* You can also check http://0.0.0.0:8001/info

## Notes

The way this is implemented has a new /upload directory, leaving the /hyperpack dir as an option to mount something into the container.

Presently the order of precedent is whenever the fastapp boots up it always starts with the provided hyperpack mounted to /hyperpack, and when a user uploads a hyperpack it uses that hyperpack instead, meaning if the container restarted for some reason after a hyperpack was uploaded it would take another upload to activate the uploaded hyperpack (even if the files are still in the upload directory). Open to changing this behaviour if we think it's better to always use what's in /upload on boot.

## Screenshots

![image](https://user-images.githubusercontent.com/4522149/190018391-60c0fd55-8d6e-4545-ae64-2037c4f7a58f.png)


## Ticket(s)

- [MLSDK-412](https://gohypergiant.atlassian.net/browse/MLSDK-412)
